share/functions/__fish_complete_users.fish

```diff
@@ -1,8 +1,10 @@
 
 function __fish_complete_users --description "Print a list of local users, with the real user name as a description"
 	if test -x /usr/bin/getent
-		getent passwd | cut -d : -f 1,5 | sed 's/:/\t/'
+		getent passwd | cut -d : -f 1,5 | string replace -r ':' \t
+	else if test -x /usr/bin/dscl
+		dscl . -list /Users RealName | string match -r -v '^_' | string replace -r ' {2,}' \t
 	else
-		__fish_sgrep -ve '^#' /etc/passwd | cut -d : -f 1,5 | sed 's/:/\t/'
+		string match -v -r '^\w*#' < /etc/passwd | cut -d : -f 1,5 | string replace ':' \t
```

Comment:
I think the intention here is to remove "functionally empty" lines, i.e. those that don't contain usable information - only comments or whitespace. So shouldn't it unmatch `^\s*#` (ie. whitespace instead of word characters)?